lib/tasks/integration.rake

```diff
@@ -0,0 +1,96 @@
+INTEGRATION_TMP_DIR     = Rails.root.join('tmp', 'integration')
+INTEGRATION_FOREMAN_PID = INTEGRATION_TMP_DIR.join('server.pid')
+INTEGRATION_FOREMAN_LOG = Rails.root.join('log', 'integration.log')
+INTEGRATION_PROCFILE    = INTEGRATION_TMP_DIR.join('Procfile')
+RUN_SINGLE_WORKER_BIN   = Rails.root.join('lib', 'workers', 'bin', 'run_single_worker.rb')
+
+namespace :integration do
+  desc "Seed the database and configure assets for integration tests"
+  task :seed => [:env, "test:vmdb:setup", "evm:foreman:seed", :compile_assets]
+
+  desc "Setup the database for integration tests"
+  task :setup => [:db_setup, "evm:foreman:setup"] do
+    mkdir_p INTEGRATION_TMP_DIR
+    rm_f    INTEGRATION_PROCFILE
+    touch   INTEGRATION_PROCFILE
+  end
+
+  desc "Run a foreman server in the background for integration tests"
+  task :run_server => [:setup] do
+    pid = File.read(INTEGRATION_FOREMAN_PID).to_i if File.exist?(INTEGRATION_FOREMAN_PID)
+
+    if pid && (Process.getpgid(pid) rescue nil).present?
+      warn "foreman process already running"
+    else
+      foreman_cmd = %W[
+        foreman start --port=#{ENV['FOREMAN_PORT'] || '3000'}
+                      --root=#{Rails.root}
+                      --procfile=#{INTEGRATION_PROCFILE}
+      ].join(" ")
+
+      pid = Process.spawn(foreman_cmd, [:out, :err] => [INTEGRATION_FOREMAN_LOG, "a"])
+      Process.detach(pid)
+      File.write(INTEGRATION_FOREMAN_PID, pid)
+    end
+  end
+
+  desc "Stop server for integration tests"
+  task :stop_server do
+    if File.exist?(INTEGRATION_FOREMAN_PID)
+      Process.kill("TERM", File.read(INTEGRATION_FOREMAN_PID).to_i)
+      rm INTEGRATION_FOREMAN_PID
+    else
+      warn "PID file for server (#{INTEGRATION_FOREMAN_PID}) doesn't exist... moving on"
+    end
+  end
+
+  desc "With a UI worker for integration tests"
+  task :with_ui => :setup do
+    ui_config = "ui: env PORT=$PORT ruby #{RUN_SINGLE_WORKER_BIN} MiqUiWorker\n"
+    File.write(INTEGRATION_PROCFILE, ui_config, :mode => "a")
+  end
+
+  # Check if a UI worker is running
+  task :ui_ready => :run_server do
+    require 'net/http'
+
+    foreman_port = ENV['FOREMAN_PORT'] || '3000'
```

Comment:
If this is configurable then I'm concerned it will conflict with the settings in cypress (which is expecting 3000 as a hardcoded default).  I think it's better to just have less knobs, so if we don't actually need the configurability right now, we should just drop it.

Reply:
> If this is configurable then I'm concerned it will conflict with the settings in cypress (which is expecting 3000 as a hardcoded default).

My assumption with this is that eventually there might be a use case where we might need to hit an external appliance for integration tests, so even `localhost` would no be correct.  That said:

> I think it's better to just have less knobs, so if we don't actually need the configurability right now, we should just drop it.

I agree from a YAGNI standpoint, and if the above does become the case in the future, we can update this then.